packages/core/lib/segment_emitter.js

```diff
@@ -1,12 +1,70 @@
 var dgram = require('dgram');
 
+var batcher = require('atomic-batcher');
 var logger = require('./logger');
 
 var DEFAULT_ADDRESS = '127.0.0.1';
 var DEFAULT_PORT = 2000;
 var PROTOCOL_HEADER = '{"format": "json", "version": 1}';
 var PROTOCOL_DELIMITER = '\n';
 
+function batchRun (work, callback) {
+  var client = dgram.createSocket('udp4');
+
+  processWork(client, work, 0, function () {
+    try {
+      client.close();
+    } finally {
+      callback();
+    }
+  });
+}
+
+function processWork (client, work, index, callback) {
+  if (index >= work.length) {
+    callback();
+    return;
+  }
+
+  sendMessage(client, work[index], function () {
+    processWork(client, work, index+1, callback);
```

Comment:
In the usage guide https://www.npmjs.com/package/atomic-batcher it seems like the batcher automatically batch the ops? What is the intention to do recursive calls with `processWork` and how does the batcher decide the proper batcher size?

Reply:
Yes, the `atomic-batcher` does batch the ops. The default batch size is as many as big as possible since the last batch. It doesn't introduce any delay in sending, it just queues up the work while existing work is in progress.

Reply:
I used recursion in `processWork` only because it was an easy way to iterate over the work using callbacks. I thought about bringing in a librayr like `async` to use `async.series` but that felt like too much. Do you have a suggestion that you prefer?